File: java/class/Construct.java
public class Construct{
    String name;
    int age;
    Construct(String name1, int age1){
        name = name1;
        age = age1;
        System.out.println("主人你好，我的名字叫"+name+"，今年我"+age+"岁了");
    }

    void hungry(){
        System.out.println("主人，"+name+"饿了^_^"); 
    }

    public static void main(String[] args){
        Construct dog1 = new Construct("小花", 2);
        dog1.hungry();
    }
}

File: java/class/extends/construct.cpp
#include <iostream>
using namespace std;

class people{
    public:
        const char *name;
        int age;
    public:
        people(const char *, int);
        void say();
};

people::people(const char *name, int age):name(name), age(age){
    cout<<"this is people construct"<<endl;
}

void people::say(){
    cout<<"people say name="<<name<<",age="<<age<<endl;
} 

class teacher: public people{
   public:
        const char *school;
   public:
        teacher(const char*, int, const char *);
        void say();
};

teacher::teacher(const char*name, int age, const char *school):people(name, age), school(school){
    cout<<"this is teacher construct"<<endl;
}

void teacher::say(){
    cout<<"teacher say name="<<name<<",age="<<age<<",school = "<<school<<endl;
}


int main(){
    people *p1 = new people("人类", 20 );
    p1->say();
    teacher *t1 = new teacher("小花", 30, "黄冈");
    t1->say();

    delete p1;
    delete t1;
    return 0;
}

